tools/install_demo_configuration.sh

```diff
@@ -408,26 +310,56 @@ $SUDO_CMD chmod +x "$OPENSEARCH_PLUGINS_DIR/opensearch-security/tools/securityad
-$SUDO_CMD chmod +x securityadmin_demo.sh
+notify "Step $((++step)): Initialize security"
+
+if [ "$initsecurity" == 0 ] && [ "$assumeyes" == 0 ]; then
+	read -r -p "Initialize Security Modules? [y/N] " response
+	case "$response" in
+	    [yY][eE][sS]|[yY])
+	        initsecurity=1
+	        ;;
+	    *)
+	        initsecurity=0
+	        ;;
+	esac
+fi
 
 if [ "$initsecurity" == 0 ]; then
-	echo "### After the whole cluster is up execute: "
+  echo "Skipping initializing OpenSearch Security based on the provided input."
+	echo "After the whole cluster is up execute: "
 	$SUDO_CMD cat securityadmin_demo.sh | tail -1
-	echo "### or run ./securityadmin_demo.sh"
-    echo "### After that you can also use the Security Plugin ConfigurationGUI"
+	echo "Or"
+	echo "run ./securityadmin_demo.sh"
+  echo "After that you can also use the Security Plugin ConfigurationGUI"
 else
-    echo "### OpenSearch Security will be automatically initialized."
-    echo "### If you like to change the runtime configuration "
-    echo "### change the files in ../securityconfig and execute: "
+  echo "plugins.security.allow_default_init_securityindex: true" | $SUDO_CMD tee -a "$OPENSEARCH_CONF_FILE" > /dev/null
+  echo "OpenSearch Security will be automatically initialized."
+  echo "If you like to change the runtime configuration "
+  echo "change the files in ../securityconfig and execute: "
 	$SUDO_CMD cat securityadmin_demo.sh | tail -1
-	echo "### or run ./securityadmin_demo.sh"
-	echo "### To use the Security Plugin ConfigurationGUI"
+	echo "or run ./securityadmin_demo.sh"
+	echo "To use the Security Plugin ConfigurationGUI"
 fi
 
-echo "### To access your secured cluster open https://<hostname>:<HTTP port> and log in with admin/admin."
-echo "### (Ignore the SSL certificate warning because we installed self-signed demo certificates)"
+# Generate securityadmin_demo.sh
+echo "#!/usr/bin/env bash" | $SUDO_CMD tee securityadmin_demo.sh > /dev/null
+echo $SUDO_CMD \""$OPENSEARCH_PLUGINS_DIR/opensearch-security/tools/securityadmin.sh"\" -cd \""$OPENSEARCH_PLUGINS_DIR/opensearch-security/securityconfig"\" -icl -key \""$OPENSEARCH_CONF_DIR/certs/kirk-key.pem"\" -cert \""$OPENSEARCH_CONF_DIR/certs/kirk.pem"\" -cacert \""$OPENSEARCH_CONF_DIR/certs/root-ca.pem"\" -nhnv | $SUDO_CMD tee -a securityadmin_demo.sh > /dev/null
+$SUDO_CMD chmod +x securityadmin_demo.sh
+
+notify "Setup complete!"
+echo -e "Single node setup successful! You can now start the cluster with bin/opensearch\n"
+echo -e "To access your secured cluster open https://<hostname>:<HTTP port> and log in with admin/admin. Example"
+echo -e "curl -k https://localhost:9200 -u admin:admin"
+echo -e "(Ignore the SSL certificate warning because we installed self-signed certificates)"
+
+echo -e "\nFollow the steps below to setup multi-node cluster"
+echo -e "1. Execute this script now on all your nodes"
```

Comment:
If this script is executed on all the nodes, will the certs have mismatched Certificate Authorities causing them to be rejected?

Reply:
Good catch. Seems like I missed updating the instructions after new changes. The certs need to be copied over to other nodes in case of multi-node cluster instead of generating afresh. Once copied the script can be executed as is and it will setup the settings without replacing the certs.

Reply:
Fixed